File: src/PM.BazaarCore.Domain/Services/CategoryService.cs
using PM.BazaarCore.Domain.Core.Services;
using PM.BazaarCore.Domain.Entities;
using PM.BazaarCore.Domain.Interfaces.Repositories;
using PM.BazaarCore.Domain.Interfaces.Services;

namespace PM.BazaarCore.Domain.Services
{
    public class CategoryService: Service<Category>, ICategoryService
    {
        public CategoryService(ICategoryRepository repository) : base(repository)
        { }
    }
}

File: app/src/app/shared/services/category.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { BaseService } from 'src/app/common/services/base.service';
import { environment } from 'src/environments/environment';
import { Category } from '../models/category';

@Injectable({
  providedIn: 'root'
})
export class CategoryService extends BaseService {

	constructor(httpClient: HttpClient) {
		super(httpClient);
	}

	public getAll(): Observable<Category[]> {
		return this.get<Category[]>(`${environment.urlBase}category`, Category);
	}   
}
